includes/multi-currency/Compatibility.php

```diff
@@ -319,6 +315,270 @@ public function convert_order_prices( $results ): array {
 		return $results;
 	}
 
+	/**
+	 * Converts the price of an addon from WooCommerce Products Add-on extension.
+	 *
+	 * @param mixed $price   The price to be filtered.
+	 * @param array $type    The type of the addon.
+
+	 * @return mixed         The price as a string or float.
+	 */
+	public function get_addons_price( $price, $type ) {
+		if ( 'percentage_based' === $type['price_type'] ) {
+			// If the addon is a percentage_based type $price is actually a percentage
+			// and doesn't need any conversion.
+			return $price;
+		}
+
+		return $this->multi_currency->get_price( $price, 'product' );
+	}
+
+	/**
+	 * Gets the product price during ajax requests from the product page.
+	 *
+	 * @param float       $price    Price to get converted.
+	 * @param int         $quantity Quantity of the product selected.
+	 * @param \WC_Product $product  WC_Product related to the price.
+	 *
+	 * @return float Adjusted price.
+	 */
+	public function get_product_calculation_price( float $price, int $quantity, \WC_Product $product ): float {
+		return $this->multi_currency->get_price( $price / $quantity, 'product' ) * $quantity;
+	}
+
+	/**
+	 * Filters the meta data for order line items so that we can properly set values in the names.
+	 *
+	 * @param array                  $meta_data A key/value for the meta data to be inserted for the line item.
+	 * @param array                  $addon     The addon being processed.
+	 * @param \WC_Order_Item_Product $item      Order item data.
+	 * @param array                  $values    Order item values.
+	 *
+	 * @return array A key/value for the meta data to be inserted for the line item.
+	 */
+	public function order_line_item_meta( array $meta_data, array $addon, \WC_Order_Item_Product $item, array $values ): array {
+
+		// If there is an add-on price, add the price of the add-on to the label name.
+		if ( $addon['price'] && apply_filters( 'woocommerce_addons_add_price_to_name', true ) ) {
+			$product = $item->get_product();
+
+			if ( $addon['price'] && 'percentage_based' === $addon['price_type'] && 0 !== $product->get_price() ) {
```

Comment:
I think the `$addon['price']` check is unnecessary here because we already check if `$addon['price']` in the `if` statement which surrounds this block of code (line 362).

Also, I got a psalm warning about a type mismatch on the `0 !== $product->get_price()` check because apparently `$product->get_price()` returns a string - it may be worth making the check `0 !== (int) $product->get_price()` to be on the safe side.

Reply:
Do you see a problem with doing it this way: `0.0 !== (float) $product->get_price()`.
Otherwise, if the price is `0.5` it turns into `0` and give a false result. 
https://github.com/Automattic/woocommerce-payments/pull/2985/files#diff-d23b7ff205ceb290f7dc0543e3845909c3361436eb26c45c672f58f9b9731522R365